eng/common/TestResources/New-TestResources.ps1

```diff
@@ -215,11 +215,18 @@ try {
     $PSBoundParameters['BaseName'] = $BaseName
 
     # Try detecting repos that support OutFile and defaulting to it
-    if (!$CI -and !$PSBoundParameters.ContainsKey('OutFile') -and $IsWindows) {
+    if (!$CI -and !$PSBoundParameters.ContainsKey('OutFile')) {
         # TODO: find a better way to detect the language
-        if (Test-Path "$repositoryRoot/eng/service.proj") {
+        if ($IsWindows -and (Test-Path "$repositoryRoot/eng/service.proj")) {
             $OutFile = $true
-            Log "Detected .NET repository. Defaulting OutFile to true. Test environment settings would be stored into the file so you don't need to set environment variables manually."
+            $isDotnet = $true
+            Log "Detected .NET repository. Defaulting OutFile to true. Test environment settings will be stored into a file so you don't need to set environment variables manually."
+        } elseif (Test-Path "$root/assets.json") {
```

Comment:
I'm not a big fan of this (or the .NET detection). Any reason to not make this a parameter to the script and set the default accordingly in the template?

Reply:
The .NET detection has always been there. This is keeping it compatible. I'll let them deal with switching to unencrypted `.env` files and deprecate their current JSON-encoded `.env`-like DPAPI-encrypted file support on their own time.

But what specifically is "this"? I need to test if the path exists before I read it in a non-terminating case; that is, if the file exists and I can successfully deserialize the check for the value I need, write the `.env` file; otherwise, keep doing what it was doing. You wanted this to be conditional. It has to be until other languages support `.env` files for this.

Reply:
I understand you want to leave the .NET code as is but for the new opt-in instead of looking for the assets.json file I would prefer that we add a new parameter to the script something along the lines of outputEnvFile and you can start passing it for the Rust usages.

Reply:
It shouldn't be something the developer has to do. It should be automatic. Or they will forget to do it and we'll get all sorts of support requests. That's how .NET on Windows support already works now. We want the users (and CI) to have the right experience by default. That's why I wrote the scripts the way they were in the first place. Complicating them by requiring more input isn't going to benefit anyone.